semantic-convention-utils/src/main/java/org/hypertrace/semantic/convention/utils/http/HttpSemanticConventionUtils.java

```diff
@@ -433,6 +450,17 @@ public static URL getNormalizedUrl(String url) throws MalformedURLException {
     return Optional.empty();
   }
 
+  private static String getProtocolFromForwarded(String value) {
```

Comment:
shall we return `Optional<String>` from here? So, we can directly check-in consumer at line number 440 (`!StringUtils.isEmpty(extractedProtoValue)`)?

`private static Optional<String> getProtocolFromForwarded(String value)`

Reply:
refactored.